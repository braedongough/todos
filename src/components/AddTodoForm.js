import React from 'react'
import { connect } from 'react-redux'
import { startAddTodo } from '../actions/todos'

export const AddTodoForm = ({ startAddTodo }) => (
    <div>
        <form
            autoComplete='off'
            onSubmit={(e) => {
                e.preventDefault()
                const description = e.target.todo.value
                startAddTodo({ description })
                e.target.todo.value = ''
            }}>
            <input
                type="text"
                placeholder="input new todo"
                name="todo"
                autoFocus={true}
            />
            <button>Submit</button>
        </form>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startAddTodo: (todoData) => dispatch(startAddTodo(todoData))
})

export default connect(undefined, mapDispatchToProps)(AddTodoForm)