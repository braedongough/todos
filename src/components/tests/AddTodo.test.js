import React from 'react'
import { shallow } from 'enzyme'
import { AddTodoForm } from '../AddTodoForm'

it('should render AddTodoForm correctly', () => {
    const wrapper = shallow(<AddTodoForm />)
    expect(wrapper).toMatchSnapshot()
})

// it('should handle onSubmit', () => {
//     const startAddTodo = jest.fn()
//     const wrapper = shallow(<AddTodoForm startAddTodo={startAddTodo} />)
//     wrapper.find('form').prop('onSubmit')()
// })