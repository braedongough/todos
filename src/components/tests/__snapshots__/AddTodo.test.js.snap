// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddTodoForm correctly 1`] = `
<div>
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      name="todo"
      placeholder="input new todo"
      type="text"
    />
    <button>
      Submit
    </button>
  </form>
</div>
`;
